tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/MatterProvision.cs

```diff
@@ -251,12 +255,36 @@ public bool CanCreateMatter(Client client)
                                 searchResult[key].ToString());
                         }
 
+                        #region Subarea of law login
+                        if (key.ToString().ToLower() == searchSettings.ManagedPropertySubAreaOfLaw2.ToLower())
+                        {
+                            subAreaOfLaw2 = searchResult[key].ToString();
+                        }
+                        if (key.ToString().ToLower() == searchSettings.ManagedPropertySubAreaOfLaw1.ToLower())
+                        {
+                            subAreaOfLaw1 = searchResult[key].ToString();
+                        }
+
                         if (key.ToString().ToLower() == searchSettings.ManagedPropertySubAreaOfLaw.ToLower())
                         {
-                            ServiceUtility.AddProperty(matterData,
-                                configuration.GetSection("Search").GetSection("SearchColumnsUIPickerForMatter").GetSection("matterSubAreaOfLaw").Key,
-                                searchResult[key].ToString());
+                            subAreaOfLaw = searchResult[key].ToString();
+                        }
+
+                        if (subAreaOfLaw2 != string.Empty && subAreaOfLaw1 != string.Empty && subAreaOfLaw != string.Empty)

```

Comment:
Modified code to sent the correct sub area of law to the client based on the matter that has been created. 
If the matter is created for SubAreaOfLevel, it will return SubAreaOfLevel
If the matter is created for SubAreaOfLevel1, it will return SubAreaOfLevel1
If the matter is created for SubAreaOfLevel2, it will return SubAreaOfLevel2